stores/flatten.go

```diff
@@ -274,5 +274,11 @@ func EncodeNonStrings(m map[string]interface{}) {
 		if vInt, ok := v.(int); ok {
 			m["shamir_threshold"] = fmt.Sprintf("%d", vInt)
 		}
+		// FlattenMetadata serializes the input as JSON and then deserializes it.
+		// The JSON unserializer treats every number as a float, so the above 'if'
+		// never applies in that situation.
+		if vFloat, ok := v.(float64); ok {
+			m["shamir_threshold"] = fmt.Sprintf("%g", vFloat)
```

Comment:
It would probably be safer to do something like: 

```suggestion
			m["shamir_threshold"] = fmt.Sprintf("%.0f", vFloat)
```

The problem is that otherwise `DecodeNonStrings` wouldn't be able to parse e.g. `1234567.0`, as it would become `1.234567e+06`.

Reply:
shamir_threshold should never be that large, but this is still better since it avoids making the result look like a floating point number!